### PR TITLE
sort trace by timestamp before calculating FMP

### PR DESCRIPTION
**Pull request by brendankenny**
re: #753 

This makes sure events in the trace are sorted by timestamp (`event.ts`) before we start digging into FMP. This probably gives us more `-1`s in extension reports but makes us more correct in the answers we report. Will have to remain this way until we figure out why trace events are timestamped the way they are or #618 with m54.

Also adds example trace from extension from #753 with weird timing for testing.

### CODE REVIEW

File: lighthouse-core/test/audits/first-meaningful-paint-test.js [hunk 1]
```diff
@@ -15,34 +15,59 @@
...
-    return Audit.audit({}).then(response => {
-      return assert.equal(response.score, -1);
+    return FMPAudit.audit({}).then(result => {
+      assert.equal(result.score, -1);
+      assert.ok(result.debugString);
     });
   });
 
   it('scores a -1 when faulty trace data is present', () => {
-    return Audit.audit({boo: 'ya'}).then(response => {
```
Comment (brendankenny) [inline lighthouse-core/test/audits/first-meaningful-paint-test.js]: this test was not actually testing faulty trace data after the trace artifact format change :)

Comment (addyosmani) [inline lighthouse-core/test/audits/first-meaningful-paint-test.js]: Newer tests are more robust. Looking good.

Comment (addyosmani) [inline lighthouse-core/test/audits/first-meaningful-paint-test.js]: and my earlier comment is caught by the tests you've written. Gotcha.

**Comment by addyosmani** [comment]
LGTM

**Pull request merged**